src/utils/daos_hdlr.c

```diff
@@ -1743,78 +1788,103 @@ file_open_dfs(file_dfs_t *file_dfs, char *file, int flags, ...)
 		} else {
 			file_dfs->fd = open(file, flags);
 		}
-		if (file_dfs->fd < 0)
-			fprintf(stderr, "daos_open failed on %s: %d\n", file, file_dfs->fd);
+		if (file_dfs->fd < 0) {
+			rc = EINVAL;
+			fprintf(stderr, "file_open failed on %s: %d\n", file, file_dfs->fd);
+		}
 	} else if (file_dfs->type == DAOS) {
 		rc = open_dfs(file_dfs, file, flags, mode);
-		if (rc != 0)
-			fprintf(stderr, "daos_open failed on %s: %d\n", file, rc);
+		if (rc != 0) {
+			fprintf(stderr, "file_open failed on %s: %d\n", file, rc);
+		}
 	} else {
-		rc = -1;
+		rc = 1;
```

Comment:
this is confusing now.
-1 is a generic error, but what is 1?
1 (errno speaking) is EPERM.
so you have to make up your mind on what you are returning.
-1 for error, or a positive errno in which case this needs to be EINVAL.

Reply:
@mchaarawi updated to EINVAL